core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -319,6 +329,7 @@ public Table loadTable(SessionContext context, TableIdentifier identifier) {
       }
     }
 
+    final TableIdentifier finalIdentifier = loadedIdent;
```

Comment:
Only instance fields should use `final` in Iceberg. This doesn't actually help the compiler do smarter things because it can be detected automatically these days (also, it isn't part of byte code). Since it doesn't help, we omit it.

Reply:
indeed this doesn't make a difference here for the lambda further below. I'll remove the `final`